recipes/paho-mqtt-cpp/all/conanfile.py

```diff
@@ -27,7 +26,7 @@ class PahoMqttCppConan(ConanFile):
     default_options = {
         "shared": False,
         "fPIC": True,
-        "ssl": True
+        "ssl": True,
```

Comment:
I feel like it deserves a comment that this is an option but has no requirement since it only changes what is being used from the C version of the library

Its a bit unusual IMO? Thoughts?

Reply:
This option should be deprecated. If you look closely at source code and upstream CML, this option it's no-op for paho-mqtt-cpp library itself. It's only used to enable few tests & samples, or for their custom FindPahoMqttC.cmake to inject an interface definition of `paho-mqtt-c`

Reply:
I also see that paho-mqtt-cpp uses its own custom `FindPahoMqttC.cmake` defining `PahoMqttC::PahoMqttC`, so file generated CMakeDeps is ignored. Let's fix that.